### PR TITLE
ashuffle: init at version 3.4.0

### PR DESCRIPTION
**Pull request by tcbravo**
##### Motivation for this change

[`ashuffle`](https://github.com/joshkunz/ashuffle) automatically shuffles the contents of an `mpd` music library when the playlist runs out of tracks. It has a grouping feature which can be used to shuffle tracks based on their tags (in other words, you can shuffle artists or albums).

I use it to keep my `mpd` playlist supplied with music, even when I am not controlling it directly.

I also added myself to the maintainers list for this package.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/ashuffle/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchgit, cmake, git, pkg-config, mpd_clientlib, meson, ninja }:
+
+stdenv.mkDerivation rec {
+  pname = "ashuffle";
+  version = "3.4.0";
+
+  src = fetchgit {
```
Comment (alexarice) [inline pkgs/applications/audio/ashuffle/default.nix:7]: is it possible to use `fetchFromGitHub` instead?

File: pkgs/applications/audio/ashuffle/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  };
+
+  nativeBuildInputs = [ pkg-config meson ];
+  buildInputs = [ cmake git mpd_clientlib ninja ];
+
+  buildFlags = ''
+    -Dbuildtype=release
+  '';
+
+  meta = {
```
Comment (alexarice) [inline pkgs/applications/audio/ashuffle/default.nix:21]: Could use `with stdenv.lib;` here and not have it below

File: pkgs/applications/audio/ashuffle/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    url = "https://github.com/joshkunz/ashuffle";
+    deepClone = true;
+    rev = "v${version}";
+    sha256 = "1k295fhhjq9n6ag605qzgxpfrr0h0xi9aggph3cy5s5i81pg7c92";
+  };
+
+  nativeBuildInputs = [ pkg-config meson ];
+  buildInputs = [ cmake git mpd_clientlib ninja ];
+
+  buildFlags = ''
```
Comment (alexarice) [inline pkgs/applications/audio/ashuffle/default.nix:17]: I believe there is a `mesonFlags` attribute (not sure on this)

File: pkgs/applications/audio/ashuffle/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchgit {
+    url = "https://github.com/joshkunz/ashuffle";
+    deepClone = true;
+    rev = "v${version}";
+    sha256 = "1k295fhhjq9n6ag605qzgxpfrr0h0xi9aggph3cy5s5i81pg7c92";
+  };
+
+  nativeBuildInputs = [ pkg-config meson ];
+  buildInputs = [ cmake git mpd_clientlib ninja ];
```
Comment (alexarice) [inline pkgs/applications/audio/ashuffle/default.nix:15]: ninja should likely be a `nativeBuildInput` as should `cmake`. As a side note it seems odd that this program uses both `cmake` and `meson`. Also `git` should probably be a `nativeBuildInput` if it is actually needed

**Review by alexarice — CHANGES_REQUESTED** [review]
Your commits should be titled:
- `maintainers: add tcbravo`
- `ashuffle: init at version 3.4.0`

File: pkgs/applications/audio/ashuffle/default.nix (hunk 4 above)
Comment (tcbravo) [reply pkgs/applications/audio/ashuffle/default.nix:15]: It doesn't build if `cmake` is in `nativeBuildInputs` - only `buildInputs`.  It appears I can move `ninja` without it affecting the build. And `git` is not needed, apparently.

I'm not sure about the difference between `buildInputs` and `nativeBuildInputs`, so sorry for the confusion.

Comment (alexarice) [reply pkgs/applications/audio/ashuffle/default.nix:15]: as a general rule `nativeBuildInputs` are needed at build time and `buildInputs` are needed when the program actually runs. This should only make a difference when cross compiling though. Is `cmake` actually needed? I expect when you move `cmake` to `nativeBuildInputs` it triggers its build hook (which probably breaks stuff)

File: pkgs/applications/audio/ashuffle/default.nix (hunk 1 above)
Comment (tcbravo) [reply pkgs/applications/audio/ashuffle/default.nix:7]: Good point, I did not see the `fetchSubmodules` option.

File: pkgs/applications/audio/ashuffle/default.nix (hunk 4 above)
Comment (tcbravo) [reply pkgs/applications/audio/ashuffle/default.nix:15]: `nix-build -A ashuffle` fails when `cmake` is completely absent, or when it is in `nativeBuildInputs`. It only passes when `cmake` is present in `buildInputs`.

Comment (alexarice) [reply pkgs/applications/audio/ashuffle/default.nix]: This seems surprising, what error do you get when cmake isn't present

Comment (tcbravo) [reply pkgs/applications/audio/ashuffle/default.nix]: Meson spits out some configuration, and then it blows up with:

```
|Executing subproject absl method cmake
|
|Did not find CMake 'cmake'
|Found CMake: NO

meson.build:37:4: ERROR: Unable to find CMake
```

So it's a submodule dependency that needs `cmake` to build.

Comment (alexarice) [reply pkgs/applications/audio/ashuffle/default.nix]: ah ok that makes sense, I still feel it should be a `nativeBuildInput`. Maybe there is someway to get cmake as a dependency without running the hook

Comment (tcbravo) [reply pkgs/applications/audio/ashuffle/default.nix]: There's a flag on `mkDerivation` called `dontUseCmakeConfigure` which makes it work.

Thanks for the help.

**Review by alexarice — APPROVED** [review]
Builds with nix-review and at least runs help page. I would still check if there is some way to move `cmake` to a native build input but otherwise looks good

@GrahamcOfBorg build

**Review by alexarice — CHANGES_REQUESTED** [review]

**Review by alexarice — APPROVED** [review]
nix-review passes on nixos

**Review by danieldk — APPROVED** [review]
LGTM.

Result of `nixpkgs-review pr 89186` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- ashuffle
</details>

Checked the binary to work.

Thanks a lot!

**Pull request merged**